doc/source/whatsnew/v1.4.0.rst

```diff
@@ -192,6 +192,11 @@ Datetimelike
 - Bug in :class:`DataFrame` constructor unnecessarily copying non-datetimelike 2D object arrays (:issue:`39272`)
 -
 
+Expressions
+^^^^^^^^^^^
+- Bug in :function:`create_valid_python_identifier` did not handle the degree sign in a backticked column name, such as \`Temp(°C)\`, using an expression to query a dataframe (:issue:`42826`)
```

Comment:
We usually try to reference only user-facing functions in whatsnew - could you instead mention `:meth:DataFrame.query` (and I suppose this could also be a bug for eval too?)

Reply:
done. might fix eval, but I can't come up with a good scenario to test.